tests/scalability/generate_brac_data/generate-brac-data.js

```diff
@@ -0,0 +1,221 @@
+  return {
+    _id: firstLevelId,
+    parent: {
+      _id: secondLevelId,
+      parent: {
+        _id: thirdLevelId
+      }
+    }
+  };
+};
+
+const pairPlaceTypesRoles = {
+  'district_hospital': 'supervisor',
+  'health_center': 'district_admin',
+  'clinic': null
+};
+
+const pairPlaceTypesNeedsUsers = {
+  'district_hospital': true,
+  'health_center': true,
+  'clinic': false
+};
+
+const pairPlaceTypesPersonSubtype = {
+  'district_hospital': 'manager',
+  'health_center': 'chw',
+  'clinic': 'member_eligible_woman'
+};
+
+const generatePerson = (type, parents, isPrimaryContact) => {
+  let subtype = pairPlaceTypesPersonSubtype[type];
+  if (type === 'clinic' && !isPrimaryContact) {
+    subtype = 'other';
+  }
+  const person = bracPersonFactory.generateBracPerson(parents, subtype);
+  createDataDoc(preconditionDataDirectory, person._id + dataExtension, JSON.stringify(person, {}, 2));
+  if (type === 'district_hospital') {
+    managers.push(person);
+  }
+  return person;
+};
+
+const generateUser = (type, placeId, userName, person, isPrimaryContact) => {
+  let roles = pairPlaceTypesRoles[type];
+  if (isPrimaryContact && type === 'district_hospital') {
+    roles = ['national_admin', 'mm-online'];
+  }
+  const personUser = bracUserFactory.generateBracUser(
+    userName,
+    roles,
+    placeId);
+  createDataDoc(preconditionDataDirectory, personUser._id + dataExtension, JSON.stringify(personUser, {}, 2));
+  const user = {
+    username: userName,
+    password: personUser.password,
+    roles: personUser.roles,
+    contact: person._id,
+    phone: person.phone,
+    place: personUser.facility_id
+  };
+  users.push(user);
+};
+
+const generateReports = (directParentPlace, place, person, isMainData) => {
+  let reportsDirectory = preconditionDataDirectory;
+  if (isMainData) {
+    reportsDirectory = path.join(mainDataDirectory, directParentPlace.contact._id);
+    createDataDirectory(mainDataDirectory, directParentPlace.contact._id);
+  }
+  if (bracPersonFactory.shouldGeneratePregnancySurvey(person)) {
+    const pregnancySurvey = bracSurvey.generateBracSurvey('pregnancy', directParentPlace, place, person);
+    createDataDoc(reportsDirectory, pregnancySurvey._id + dataExtension, JSON.stringify(pregnancySurvey, {}, 2));
+  }
+  if (bracPersonFactory.shouldGenerateAssessmentSurvey(person)) {
+    const assesmentSurvey = bracSurvey.generateBracSurvey('assesment', directParentPlace, place, person);
+    createDataDoc(reportsDirectory, assesmentSurvey._id + dataExtension, JSON.stringify(assesmentSurvey, {}, 2));
+    const assesmentFollowUpSurvey = bracSurvey.generateBracSurvey(
+      'assesment_follow_up', directParentPlace, place, person);
+    createDataDoc(reportsDirectory, assesmentFollowUpSurvey._id + dataExtension,
+      JSON.stringify(assesmentFollowUpSurvey, {}, 2));
+  }
+};
+
+const generateHierarchy = (type, placeName, numberOfPersons,
+  parentFirstLevelId, parentSecondLevelId, directParentPlace) => {
```

Comment:
I think that we can extract the lineage from the "directParentPlace" object itself, and no need to pass them as extra parameters. 
```javascript
const generateHierarchy = async (type, placeName, parentPlace, numberOfPersons) => {
....
  const placeLineage = { _id: parent._id, parent: parentPlace.parent };
  const place = bracPlaceFactory.generateBracPlace(placeName, type, placeLineage);
  const personLineage = { _id: place._id, parent: place.parent };
  .....
}
```